payjoin/src/uri.rs

```diff
@@ -2,27 +2,49 @@ use std::borrow::Cow;
 use std::convert::TryFrom;
 
 use bitcoin::address::{Error, NetworkChecked, NetworkUnchecked};
-use bitcoin::Network;
+use bitcoin::{Address, Amount, Network};
 use url::Url;
 
+/// Payjoin enum represents a payjoin object.
+///
+/// Used in building a `bip21::Uri` with `uri::PayjoinParams`.
 pub enum Payjoin {
+    /// Represents a supported payjoin endpoint.
```

Comment:
```suggestion
    /// A supported payjoin endpoint.
```

&c for the following